### PR TITLE
Loginout : Add border and color block support

### PR DESCRIPTION
**Pull request by akasunil**
## What?

Add border and color block support to the LoginOut block.

Part of 
https://github.com/WordPress/gutenberg/issues/43247
https://github.com/WordPress/gutenberg/issues/43245
## Why?
The block is missing border and color support.

## How?

Adds the border and color block support in block.json

## Testing Instructions

- Go to Global Styles setting ( under appearance > editor > styles > edit styles > blocks )
- Make sure that `Login/out`  block's styles is configurable via Global Styles
- Verify that Global Styles are applied correctly in the editor and frontend
- Edit template/page, Add `Login/out`  block and Apply the styles
- Verify that block styles take precedence over global styles
- Verify that block styles display correctly in both the editor and frontend

## Screenshots or screencast 
 
[Blog-Home-‹-Template-‹-gutenberg-‹-Editor-—-WordPress (6).webm](https://github.com/user-attachments/assets/b80305f0-a746-461a-a005-bca63849795a)

### CODE REVIEW

File: packages/block-library/src/loginout/block.json [hunk 1]
```diff
@@ -43,6 +43,18 @@
...
+		"__experimentalBorder": {
+			"radius": true,
+			"color": true,
+			"width": true,
+			"style": true,
+			"__experimentalDefaultControls": {
+				"radius": false,
+				"color": false,
+				"width": false,
+				"style": false
```
Comment (aaronrobertshaw) [inline packages/block-library/src/loginout/block.json]: I believe generally when the border supports have been adopted their controls are being displayed by default.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for putting this PR together @akasunil, it's great to see more blocks adopting the available block supports! 👍 

The border support is testing well for me on the login/out block.

✅ Editor
✅ Frontend
✅ Logged in and out
✅ Theme.json/global styles

I think we might want to make the border controls all display by default though. This has been the general approach to date for blocks adopting this support and would further improve the consistency of design tools in Gutenberg.

What do you think?

File: packages/block-library/src/loginout/block.json (hunk 1 above)
Comment (akasunil) [reply packages/block-library/src/loginout/block.json]: Agreed.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for iterating on this 👍 

I missed double checking the login/out block within different contexts last time I reviewed, such as within a Group block with custom content width. I think this block might also benefit from applying `box-sizing: border-box`.

Apologies that this is a bit of a recurring theme on your block support adoption PRs. If it is any consolation, there are a few of my own I'm going to need to go and revisit as well. 

| Without border-box sizing | With border-box sizing |
|---|---|
| Screenshot 2024-07-17 at 5 10 52 PM | Screenshot 2024-07-17 at 5 11 03 PM |

In the above screenshots, you can see how padding and borders push the login/out block beyond the bounds that the parent Group is trying to enforce on its content.

**Comment by akasunil** [comment]
@aaronrobertshaw I apologize for not noticing this before. Since this PR only add supports of border styles, my attention was solely on those.

**Comment by akasunil** [comment]
I think we should check all blocks with layout supports. `box-sizing: border-box` style could be included for each of those blocks

**Comment by aaronrobertshaw** [comment]
> I apologize for not noticing this before. Since this PR only add supports of border styles, my attention was solely on those.

No apologies needed 🙂 

It's easy to miss that adding borders can impact the actual width of a block depending on its context. I know I did to start with as well.

**Comment by akasunil** [comment]
@aaronrobertshaw 
I have also added color support in this block, and disabled text color support as this block only have links, followed your feedback from tag cloud block https://github.com/WordPress/gutenberg/pull/63592#discussion_r1680558880

I have added video recording of block styles testing in PR description

**Review by aaronrobertshaw — APPROVED** [review]
Thanks for the iteration on this @akasunil 👍 

✅ Border and color global styles work
✅ Block instance styles for border and color apply correctly (except issue below)
✅ Box sizing allows parent block to enforce widths

The one issue I noticed was not being able to override a gradient from global styles with a simple background color on the block instance. This is a broader, preexisting problem and not introduced here. It can be a separate follow-up.

LGTM 🚢 


https://github.com/user-attachments/assets/682010b1-1d60-4af6-9ab0-6857810b42dd

**Comment by akasunil** [comment]
> The one issue I noticed was not being able to override a gradient from global styles with a simple background color on the block instance. This is a broader, preexisting problem and not introduced here. It can be a separate follow-up.

Looks like its issue of different css property used for simple and gradient background, it uses `background-color` for the background color and `background` for the gradient. Gradient from block styles override gradient from global styles.

[Blog-Home-‹-Template-‹-gutenberg-‹-Editor-—-WordPress.webm](https://github.com/user-attachments/assets/ee84d2a0-2f26-496a-8e5f-5adbb327c25b)

Will create separate issue for this if its not already created.

**Pull request merged**